src/test/java/com/google/api/generator/gapic/composer/ResourceNameHelperClassComposerTest.java

```diff
@@ -227,5 +227,62 @@ public void generateResourceNameClass() {
           + "  public Builder toBuilder() {\n"
           + "    return new Builder(this);\n"
           + "  }\n"
+          + "\n"
+          + "  public static FoobarName of(String project, String foobar) {\n"
+          + "    return newProjectFoobarBuilder().setProject(project).setFoobar(foobar).build();\n"
+          + "  }\n"
+          + "\n"
+          + "  @BetaApi(\"The static create methods are not stable yet and may be changed in the"
+          + " future.\")\n"
+          + "  public static FoobarName ofProjectFoobarBuilder() {\n"
+          + "    return newProjectFoobarBuilder().setProject(project).setFoobar(foobar).build();\n"
```

Comment:
below `of***Builder` method all has arguments of the setter tokens, is this intended here that we will construct the Builder from the global variable `project` and `foobar` instead of passing them in?

Reply:
Fixed.